### PR TITLE
Debounce new requests changes coming from the callback

### PR DESCRIPTION
**Pull request by jwallet**
If you click rapidly on make requests in the example, you will find out that the app freezes (even if the logger is paused). The amount of re-render is just crazy for the setCallback subscriber in NetworkLogger. 

- I added a `refreshRate` to debounce the callback, so the subscriber do not get 50 events in the same milisecond, default to 50milliseconds. This option was added to StartNetworkLoggingOptions
- I added a `updatedAt` property on the Request class, so you can publish through the callback only when requests are being updated or added, clearing the requests array will reset the `latestRequestUpdatedAt` to `0`
- I spreaded the array before sending it through the callback to prevent mutation of requests for all subscribers

##### before (i clicked like 30 times on pause -- not responding)
[before](https://github.com/alexbrazier/react-native-network-logger/assets/23088305/d8fc3fb3-ebdd-4719-b259-9a3cbd1684e5)

##### after
[after](https://github.com/alexbrazier/react-native-network-logger/assets/23088305/85f54d03-bc4b-46d6-ab48-bfcd0cca8dd2)

### CODE REVIEW

File: src/components/NetworkLogger.tsx [hunk 1]
```diff
@@ -43,7 +41,7 @@ const NetworkLogger: React.FC<Props> = ({ theme = 'light', sort = 'desc' }) => {
 
   useEffect(() => {
     logger.setCallback((updatedRequests: NetworkRequestInfo[]) => {
-      setRequests(sortRequests(updatedRequests, sort));
+      setRequests(updatedRequests);
```
Comment (jwallet) [inline src/components/NetworkLogger.tsx:44]:
```suggestion
      setRequests([...updatedRequests]);
```

**Comment by alexbrazier** [comment]
I tried to resolve the conflicts here but looks like I don't have permission to push

**Comment by jwallet** [comment]
Yes the company settings don't allow that, I will have to merge conflicts myself

**Review by alexbrazier — APPROVED** [review]
Thanks, looks like quite a big improvement

**Pull request merged**